### ISSUE TITLE
Rust compiler doesn't warns while accessing invalid indexes in a slice.

### ISSUE THREAD

**Issue by Ishtmeet-Singh** [rust-lang/rust#98272]
Since slices are known to have length known at all time, shouldn't it be able to figure out whether the 'user' is trying to access an index that's out of bounds?

```rs
let my_str: &str = "Hello world";
println!("{}", &my_str[0..300]);
```

This compiles just fine - shouldn't it suppose to fail and warn us about that?

**Comment by eggyal** [comment]
The length of a slice is known at runtime.  It is not, generally, known at compile-time.

Of course, in some situations the slice length could be statically determined (e.g. where the slice is of an array or, as in your example, a string literal).  But this isn't something the complier does actually check.

**Comment by Urgau** [comment]
But this clippy does for simple cases: [out_of_bounds_indexing](https://rust-lang.github.io/rust-clippy/master/#out_of_bounds_indexing).
